### PR TITLE
Update performance-tips.md

### PR DESCRIPTION
**Pull request by StephenVavasis**
I added a paragraph concerning something that surprised me about `convert`.

### CODE REVIEW

File: doc/src/manual/performance-tips.md [hunk 1]
```diff
@@ -477,6 +477,12 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
 an annotation like this has the added benefit that it will raise a run-time error if the
 value is not of the expected type, potentially catching certain bugs earlier.
 
+A slightly more generic way to annotate the type of `x` is `x = convert(Int32,a[1])::Int32`.  The
```
Comment (kshyatt) [inline doc/src/manual/performance-tips.md]: Perhaps rephrase this to "A slightly more generic way to annotate the type of `x` is shown by the example of...". It took me a second to realize you are referring to the `::Int32` bit and the fact that `convert` is involved is somewhat incidental (if I've understood correctly).

Comment (StephenVavasis) [inline doc/src/manual/performance-tips.md]: This paragraph is supposed to make two points: First, to increase genericity, a `convert` function can be used rather than a direct annotation.  In other words, the use of `convert` allows more possible types for `a[1]`.  The second point (which is the point that surprised me) is that if `convert` is used in this manner, it needs a type annotation else it is type-unstable.

Comment (kshyatt) [inline doc/src/manual/performance-tips.md]: The second part comes through pretty well but your response to me here, if worked into what you already have, would make it a lot clearer I think. Thanks for adding this!

**Review by kshyatt — APPROVED** [review]
LGTM! Thanks for bearing with me @StephenVavasis

File: doc/src/manual/performance-tips.md [hunk 2]
```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
 an annotation like this has the added benefit that it will raise a run-time error if the
 value is not of the expected type, potentially catching certain bugs earlier.
 
+In the case that the type of `a[1]` is not known precisely, `x` can be declared via
+`x = convert(Int32,a[1])::Int32`.  The
+use of the `convert` function allows `a[1]` to be any object convertible to an `Int32` (such as `UInt8`), thus
```
Comment (fredrikekre) [inline doc/src/manual/performance-tips.md]: Perhaps worth adding a reference to `convert`? E.g. ``[`convert`](@ref)``

File: doc/src/manual/performance-tips.md [hunk 3]
```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
...
+increasing the genericity of the code by loosening the type requirement.
+Notice that the `convert` function itself needs a type annotation in this context in order to achieve type stability.
+This is because the compiler cannot deduce the type of the return value of a function, even
+`convert`, unless the types of all its arguments are known.
```
Comment (fredrikekre) [inline doc/src/manual/performance-tips.md]: Perhaps "...the types of all the arguments...", since "its" could refer to the compiler in this sentence?

File: doc/src/manual/performance-tips.md [hunk 4]
```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
 an annotation like this has the added benefit that it will raise a run-time error if the
 value is not of the expected type, potentially catching certain bugs earlier.
 
+In the case that the type of `a[1]` is not known precisely, `x` can be declared via
+`x = convert(Int32,a[1])::Int32`.  The
```
Comment (fredrikekre) [inline doc/src/manual/performance-tips.md]: Double space before "The" ?

File: doc/src/manual/performance-tips.md [hunk 5]
```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
...
+use of the `convert` function allows `a[1]` to be any object convertible to an `Int32` (such as `UInt8`), thus
+increasing the genericity of the code by loosening the type requirement.
+Notice that the `convert` function itself needs a type annotation in this context in order to achieve type stability.
```
Comment (fredrikekre) [inline doc/src/manual/performance-tips.md]: Perhaps "Notice that `convert` itself..." ?

File: doc/src/manual/performance-tips.md (hunk 2 above)
Comment (nalimilan) [inline doc/src/manual/performance-tips.md]: Also wrap lines at 92 chars.

File: doc/src/manual/performance-tips.md [hunk 6]
```diff
@@ -477,6 +477,14 @@ Here, we happened to know that the first element of `a` would be an [`Int32`](@r
 an annotation like this has the added benefit that it will raise a run-time error if the
 value is not of the expected type, potentially catching certain bugs earlier.
 
+In the case that the type of `a[1]` is not known precisely, `x` can be declared via
+`x = convert(Int32,a[1])::Int32`. The use of the [`convert`](@ref) function allows `a[1]`
+to be any object convertible to an `Int32` (such as `UInt8`), thus increasing the genericity
+of the code by loosening the type requirement. Notice that `convert` itself needs a type
```
Comment (fredrikekre) [inline doc/src/manual/performance-tips.md]: Trailing space which causes CI to fail, same on the next 2 lines.

**Review by nalimilan — APPROVED** [review]
Thanks!

**Comment by nalimilan** [comment]
@tkelman Would have been nice to remove mentions of our usernames from the commit message.
